tests/test_fft.py

```diff
@@ -20,294 +19,178 @@ def test_n_bins_odd():
-    # in the given time interval
-    frequency = num_periods * samplingrate / num_samples
-    times = np.arange(0, num_samples) / samplingrate
-
-    signal_time = 1 * np.cos(2 * np.pi * frequency * times)
-    signal_spec = np.fft.rfft(signal_time, n=num_samples, axis=-1)
-    transformed_signal_time = np.fft.irfft(signal_spec, n=num_samples, axis=-1)
-
-    npt.assert_allclose(signal_time, transformed_signal_time, rtol=1e-10)
-
-
-def test_fft_orthogonality_sine_odd_lib():
-    num_samples = 2**10+3
-    frequency = 10e3
-    samplingrate = 40e3
-    num_periods = np.floor(num_samples / samplingrate * frequency)
-    # round to the nearest frequency resulting in a fully periodic sine signal
-    # in the given time interval
-    frequency = num_periods * samplingrate / num_samples
-    times = np.arange(0, num_samples) / samplingrate
-
-    signal_time = 1 * np.cos(2 * np.pi * frequency * times)
-    signal_spec = np.fft.rfft(signal_time, n=num_samples, axis=-1)
-    transformed_signal_time = np.fft.irfft(signal_spec, n=num_samples, axis=-1)
-
-    npt.assert_allclose(signal_time, transformed_signal_time, rtol=1e-10)
-
-
-def test_fft_orthogonality_noise_even_lib():
-    n_samples = 2**18
-    np.random.seed(450)
-    signal_time = np.random.normal(0, 1, n_samples)
-    signal_spec = np.fft.rfft(signal_time, n=n_samples, axis=-1)
-    transformed_signal_time = np.fft.irfft(signal_spec, n=n_samples, axis=-1)
-
-    npt.assert_allclose(signal_time, transformed_signal_time, rtol=1e-8)
-
-
-def test_fft_orthogonality_noise_odd_lib():
-    n_samples = 2**18+1
-    np.random.seed(450)
-    signal_time = np.random.normal(0, 1, n_samples)
-    signal_spec = np.fft.rfft(signal_time, n=n_samples, axis=-1)
-    transformed_signal_time = np.fft.irfft(signal_spec, n=n_samples, axis=-1)
-
-    npt.assert_allclose(signal_time, transformed_signal_time, rtol=1e-10)
-
-
-def test_fft_orthogonality_sine_even_np(fft_lib_np):
-    num_samples = 2**10
-    frequency = 10e3
-    samplingrate = 40e3
-    num_periods = np.floor(num_samples / samplingrate * frequency)
-    # round to the nearest frequency resulting in a fully periodic sine signal
-    # in the given time interval
-    frequency = num_periods * samplingrate / num_samples
-    times = np.arange(0, num_samples) / samplingrate
-
-    signal_time = 1 * np.cos(2 * np.pi * frequency * times)
-    signal_spec = fft.rfft(signal_time, num_samples, samplingrate, 'rms')
+def test_fft_orthogonality_sine_even_np(sine, fft_lib_np):
+    signal_spec = fft.rfft(
+        sine.time, sine.n_samples, sine.sampling_rate, sine.fft_norm)
     transformed_signal_time = fft.irfft(
-        signal_spec, num_samples, samplingrate, 'rms')
-
-    npt.assert_allclose(signal_time, transformed_signal_time, rtol=1e-10)
-
+        signal_spec, sine.n_samples, sine.sampling_rate, sine.fft_norm)
+    npt.assert_allclose(
+        transformed_signal_time, sine.time,
+        rtol=1e-10, atol=10*np.finfo(float).eps)
```

Comment:
missing whitespaces (flake8: E226)
... this occurs in the majority of the following test functions in this file.

Reply:
Mmmm still missing the whitespace around math operators. In the end this is not important at all. On the other hand why don't we share all the same flake-ignore-list with the CI-server to avoid this?

Reply:
I did not change it here as it appears in `test_fft.py`, where already other flake8 warnings appear in the code I did not touch... I would suggest to open a different issues on the flake8 problems appearing in various files?

Reply:
From https://flake8.pycqa.org/en/2.5.5/config.html:
"If the ignore option is not in the configuration and not in the arguments, only the error codes E123/E133, E226 and E241/E242 are ignored (see the warning and error codes)."

I'll approve that and leave this point to the next meeting.